subsys/net/l2/openthread/Kconfig

```diff
@@ -189,6 +189,23 @@ endchoice
 config OPENTHREAD_MBEDTLS
 	bool
 	select MBEDTLS
+	imply MBEDTLS_PSA_CRYPTO_C
+	select MBEDTLS_USE_PSA_CRYPTO
```

Comment:
This is [not strictly required ](https://github.com/zephyrproject-rtos/zephyr/blob/4e4ec9c1e201a49abad9b1207fe0d4872306eeaf/modules/mbedtls/Kconfig.tls-generic#L552)as it's `default y if MBEDTLS_PSA_CRYPTO_CLIENT`. In other words you will have `CONFIG_MBEDTLS_USE_PSA_CRYPTO=y` if either  `CONFIG_MBEDTLS_PSA_CRYPTO_C=y` or `CONFIG_BUILD_WITH_TFM=y`, and one of the 2 is true for sure based on the comment above.